pkg/asset/installconfig/ovirt/credentials.go

```diff
@@ -1,92 +1,231 @@
-			ovirtURL.Scheme,
-			ovirtURL.Host)
-
-		err = survey.AskOne(&survey.Multiline{
-			Message: "oVirt certificate bundle",
-			Help:    fmt.Sprintf("The oVirt certificate bundle can be downloaded from %s.", pemURL),
-		},
-			&c.CABundle,
-			survey.ComposeValidators(survey.Required))
+
+		err = askPassword(&c)
 		if err != nil {
-			return c, err
+			loginAttempts = loginAttempts - 1
+			logrus.Debugf("login attempts now: %d", loginAttempts)
+			if loginAttempts == 0 {
+				return c, err
+			}
+		} else {
+			break
 		}
-	} else {
-		logrus.Warning("Communication with the oVirt engine will be insecure.")
 	}
+	return c, nil
+}
+
+// engineSetup will ask users: FQDN, execute validations and about
+// the credentials. In case of failure, returns Config and error
+func engineSetup() (Config, error) {
+	engineConfig := Config{}
+	httpResource := clientHTTP{}
 
-	err = survey.Ask([]*survey.Question{
+	err := survey.Ask([]*survey.Question{
 		{
 			Prompt: &survey.Input{
-				Message: "oVirt engine username",
-				Help:    "The user must have permissions to create VMs and disks on the Storage Domain with the same name as the OpenShift cluster.",
-				Default: "admin@internal",
+				Message: "Engine FQDN[:PORT]",
+				Help:    "The Engine FQDN[:PORT] (engine.example.com:443)",
 			},
 			Validate: survey.ComposeValidators(survey.Required),
 		},
-	}, &c.Username)
+	}, &engineConfig.FQDN)
 	if err != nil {
-		return c, err
+		return engineConfig, err
 	}
+	logrus.Debug("engine FQDN: ", engineConfig.FQDN)
 
-	err = survey.Ask([]*survey.Question{
-		{
-			Prompt: &survey.Password{
-				Message: "oVirt engine password",
-				Help:    "",
-			},
-			Validate: survey.ComposeValidators(survey.Required, authenticated(&c)),
-		},
-	}, &c.Password)
+	// By default, we set Insecure true
+	engineConfig.Insecure = true
+
+	// Set c.URL with the API endpoint
+	engineConfig.URL = fmt.Sprintf("https://%s/ovirt-engine/api", engineConfig.FQDN)
+	logrus.Debug("Engine URL: ", engineConfig.URL)
+
+	// Start creating clientHTTP struct for checking if Engine FQDN is responding
+	httpResource.skipVerify = true
+	httpResource.urlAddr = engineConfig.URL
+	err = httpResource.checkURLResponse()
 	if err != nil {
-		return c, err
+		return engineConfig, err
 	}
 
-	return c, nil
+	// Set Engine PEM URL for Download
+	engineConfig.PemURL = fmt.Sprintf(
+		"https://%s/ovirt-engine/services/pki-resource?resource=ca-certificate&format=X509-PEM-CA",
+		engineConfig.FQDN)
+	logrus.Debug("PEM URL: ", engineConfig.PemURL)
+
+	// Create tmpFile to store the Engine PEM file
+	tmpFile, err := ioutil.TempFile(os.TempDir(), "engine-")
+	if err != nil {
+		return engineConfig, err
+	}
+	defer os.Remove(tmpFile.Name())
+
+	// Download PEM
+	httpResource.saveFilePath = tmpFile.Name()
+	httpResource.skipVerify = true
+	httpResource.urlAddr = engineConfig.PemURL
+	err = httpResource.downloadFile()
+	if err != nil {
+		logrus.Warning("cannot download PEM file from Engine!", err)
```

Comment:
So a failure to download the pem means we just ignore it from now on?

Reply:
If failure customer will be able to continue or not (with a big warning in their screen.) bye CTRL C

Reply:
This should be a question if to proceed. Ctrl C means they don't know what's the state of their RHV/oVirt cluster after that. They have resources created or not. That's not a desired experience.

Reply:
> This should be a question if to proceed. Ctrl C means they don't know what's the state of their RHV/oVirt cluster after
> that. They have resources created or not. That's not a desired experience.

fell free to open a card on that... we are talking here 4.5 branch not master. The patch was there in master for review and get approved.

Reply:
> fell free to open a card on that... we are talking here 4.5 branch not master. The patch was there in master for review and get approved.

in that regard it's helpful if the PR links to the upstream PR so that it's clear whether or not this was previously discussed. If the bot were used to backport via `/cherrypick release-4.5` the bot would've linked to the upstream PR.